Error when password has the character !

### Problem

It seems that diagnostics will fail if the password has a `!` character.

### Steps to reproduce

```
./diagnostics.sh --type api -u another --password --ssl --noVerify --bypassDiagVerify --host my-host  --port 9200
```

Password: `changeme!0`

The collected password is thus incorrect and we get:

```
Using /Library/Java/JavaVirtualMachines/adoptopenjdk-11.jdk/Contents/Home/bin/java as Java Runtime
Using -Xms256m -Xmx2000m  for options.
Processing diagnosticInputs...
Elasticsearch user password: *********
[ERROR] Could not expand event
java.lang.IllegalArgumentException: !0: event not found
        at jline.console.ConsoleReader.expandEvents(ConsoleReader.java:855)
        at jline.console.ConsoleReader.finishBuffer(ConsoleReader.java:715)
        at jline.console.ConsoleReader.accept(ConsoleReader.java:2025)
        at jline.console.ConsoleReader.readLine(ConsoleReader.java:2698)
        at org.beryx.textio.jline.JLineTextTerminal.read(JLineTextTerminal.java:240)
        at org.beryx.textio.InputReader.readWithPrompt(InputReader.java:460)
        at org.beryx.textio.InputReader.lambda$read$3(InputReader.java:405)
        at org.beryx.textio.TextTerminal.applyWithPropertiesConfigurator(TextTerminal.java:243)
        at org.beryx.textio.InputReader.executeWithTerminal(InputReader.java:484)
        at org.beryx.textio.InputReader.read(InputReader.java:403)
        at org.beryx.textio.InputReader.read(InputReader.java:391)
        at com.elastic.support.rest.ElasticRestClientInputs.parseInputs(ElasticRestClientInputs.java:111)
        at com.elastic.support.diagnostics.DiagnosticInputs.parseInputs(DiagnosticInputs.java:221)
        at com.elastic.support.diagnostics.DiagnosticApp.main(DiagnosticApp.java:26)


Creating temp directory: /tmp/api-diagnostics
Configuring log file.
Getting Elasticsearch Version.
Could not retrieve the Elasticsearch version - unable to continue. Status: 401  Reason: Authentication failure. Invalid login credentials.
Closing logger.
Archiving diagnostic results.
Archive: /tmp/api-diagnostics-20200401-073324.tar.gz was created
```

### Workaround

The workaround is to ask the customer to use a user that does not have a `!` in the password.

--- a/pom.xml
+++ b/pom.xml
@@ -4,7 +4,7 @@
 
     <groupId>com.elasticsearch</groupId>
     <artifactId>support-diagnostics</artifactId>
-    <version>8.0.3</version>
+    <version>8.0.4</version>
     <packaging>jar</packaging>
     <name>Support Diagnostics Utilities</name>
     <properties>
@@ -118,8 +118,19 @@
                     <groupId>org.slf4j</groupId>
                     <artifactId>slf4j-api</artifactId>
                 </exclusion>
+                <exclusion>
+                    <groupId>jline</groupId>
+                    <artifactId>jline</artifactId>
+                </exclusion>
             </exclusions>
         </dependency>
+
+        <dependency>
+            <groupId>jline</groupId>
+            <artifactId>jline</artifactId>
+            <version>2.14.6</version>
+        </dependency>
+
 
         <dependency>
             <groupId>org.junit.platform</groupId>

--- a/src/main/java/com/elastic/support/util/ResourceCache.java
+++ b/src/main/java/com/elastic/support/util/ResourceCache.java
@@ -1,11 +1,14 @@
 package com.elastic.support.util;
 
 import com.elastic.support.rest.RestClient;
+import jdk.tools.jlink.internal.Jlink;
+import jline.console.ConsoleReader;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.beryx.textio.TextIO;
 import org.beryx.textio.TextIoFactory;
 import org.beryx.textio.TextTerminal;
+import org.beryx.textio.jline.JLineTextTerminal;
 
 import java.io.Closeable;
 import java.util.concurrent.ConcurrentHashMap;
@@ -18,6 +21,14 @@ public class ResourceCache{
 
     public static final TextIO textIO = TextIoFactory.getTextIO();
     public static final TextTerminal terminal = textIO.getTextTerminal();
+
+    static {
+        if(terminal instanceof JLineTextTerminal){
+            JLineTextTerminal jltt = (JLineTextTerminal)terminal;
+            ConsoleReader reader = jltt.getReader();
+            reader.setExpandEvents(false);
+        }
+    }
 
     public static boolean resourceExists(String name){
         return resources.containsKey(name);

